packages/SwingSet/docs/vat-upgrade.md

```diff
@@ -63,6 +63,8 @@ During the upgrade phase, v2 code is obligated to re-define all durable Kinds cr
 
 As a special case, the root object returned from v2's `buildRootObject()` is automatically associated with exportID `o+0` (see [How Liveslots Uses the Vatstore](../../swingset-liveslots/src/vatstore-usage.md#counters)) and is therefore also obligated to support the same methods as its predecessor. This means that the root object is effectively always durable, and should not be explicitly persisted.
 
+To be precise, the root object *must* be an "ephemeral" object, i.e. created with `Far` or `makeExo()`. It cannot be a virtual or durable object (created with a maker returned by `defineKind` or `defineDurableKind`, or the vat-data convenience wrappers like `prepareExo` or `prepareSingleton`). This ensures that the root object's identity is stable across upgrades.
```

Comment:
Likewise here.
```suggestion
To be precise, the root object *must* be an "ephemeral" object as documented in [swingset-liveslots](https://github.com/Agoric/agoric-sdk/blob/master/packages/swingset-liveslots/docs/liveslots.md#buildrootobject) (e.g., created with `Far` or `makeExo()`).
```

Reply:
This one I'm inclined to leave in, because implementing upgradability is exactly the time where people are most likely to want to use `prepareSingleton`. I'll add the cross link, though.